text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,502 @@
+
+* New function for orchestrating asset deployment and infrastructure deployment for stack sets. More on asset deployment describe in the Lambda assets section.
+
+### [3.3] deploy-stack-set.ts
+This will be a new file which will contain the logic and api calls for deploying stack sets. Stack sets differ not only in API calls, but also a few service caveats:
+* No concept of change sets
+* Stack Set Operation Management
+
+### General Flow:
+
+![Architecture Diagram](../images/self-managed-stack-sets/low_level_stack_set_deploy.jpg)
+* Check if the stack set exists
+  * If not, then create
+* If the stack set does exist, check if there are any stack set operations in progress and wait until complete
+* When the stack set is ready to accept operations, call update stack set
+  * When calling update stack set API, no accounts or regions will be supplied, which will trigger an update on the stack set itself and all stack set instances
+* Wait for the update stack set operation to complete
+
+## [4] File Assets Management
+In a normal stack deployment, artifacts are defined within specific AWS resource constructs such as lambda or just the assets construct in general. They are published to S3 and referenced in the generated template by parameters (LegacyStackSynthesizer) or direct injection (DefaultStackSynthesizer). For Stack Sets, the same behavior is ideal, but there are a few more caveats:
+
+* Permissions for stack set child accounts to access assets
+* Assets in S3 must exist in every region the stack set deploys to
+* The generated template must contain generic S3 references for account/region since the same template should be used across all child accounts (cdk-<>-assets-${AWS::AccountId}-${AWS::Region})
+* In Scope:
+  * File Assets
+
+### High Level Architecture
+![Architecture Diagram](../images/self-managed-stack-sets/file_asset_high_level_arch.jpg)
+* Before Deploy:
+  * Deploy custom bootstrap infrastructure to all regions or regions stack set will be deploying to in management account
+* During CDK deploy:
+  * Assets are published to CDK Toolkit staging buckets in Management account
+  * Stack set create/update is executed in Management Account
+  * Stack set instance retrieves artifacts from management Account staging bucket
+
+### [4.1] Bootstrapping
+
+CDK apps that utilize assets require “Bootstrap” infrastructure for storing and publishing assets. This can either be done using the cdk bootstrap command which will deploy a default stack with all the required infrastructure or a customized template. 
+
+### Permissions
+Since stack set instances are deployed outside the CDK context and can be deployed to “external” accounts, child accounts need permissions to retrieving assets from the centralized staging bucket in the management account. This can be achieved by adding a bucket policy to the staging buckets:
+
+```json
+{
```

Comment:
Can you show the entire resource that you're adding to the bootstrap template here?

Reply:
Still no resource here.

Reply:
Added